src/sentry/models/useremail.py

```diff
@@ -1,37 +1,45 @@
+from __future__ import annotations
+
 from collections import defaultdict
 from datetime import timedelta
-from typing import TYPE_CHECKING
+from typing import TYPE_CHECKING, Iterable, Mapping
 
 from django.conf import settings
 from django.db import models
+from django.db.models import QuerySet
 from django.utils import timezone
 from django.utils.crypto import get_random_string
 from django.utils.translation import ugettext_lazy as _
 
-from sentry.db.models import FlexibleForeignKey, Model, sane_repr
-from sentry.db.models.manager import BaseManager
+from sentry.db.models import BaseManager, FlexibleForeignKey, Model, sane_repr
 
 if TYPE_CHECKING:
-    from sentry.models import User
+    from sentry.models import Organization, User
 
 CHARACTERS = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
 
 
 class UserEmailManager(BaseManager):
-    def get_for_organization(self, organization):
+    def get_for_organization(self, organization: Organization) -> QuerySet:
         return self.filter(user__sentry_orgmember_set__organization=organization)
 
-    def get_emails_by_user(self, organization):
+    def get_emails_by_user(self, organization: Organization) -> Mapping[User, Iterable[str]]:
         emails_by_user = defaultdict(set)
         user_emails = self.get_for_organization(organization).select_related("user")
         for entry in user_emails:
             emails_by_user[entry.user].add(entry.email)
         return emails_by_user
 
-    def get_primary_email(self, user: "User") -> "UserEmail":
+    def get_primary_email(self, user: User) -> UserEmail:
         user_email, _ = self.get_or_create(user=user, email=user.email)
         return user_email
 
+    def get_users_by_id(self, email: str) -> Mapping[int, User]:
```

Comment:
I see this moved but I still don't see it being used anywhere? What's the purpose?

Reply:
It looks like the usage got lost in a rebase I'll move it to it's own PR.